{Release} Upgrade to Azure CLI 2.27.0 (#19052)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.27.0
++++++

**ACR**

* [BREAKING CHANGE] `az acr connected-registry install info`: Add a new required parameter `--parent-protocol`. (#18954)
* [BREAKING CHANGE] `az acr connected-registry install renew-credentials`: Add a new required parameter `--parent-protocol`. (#18954)
* `az acr import`: Support new parameter `--no-wait` (#18572)

> * Fix some bugs on uncovered cmds (#18786)

Reviewer: ```suggestion
* Fix the Python SDK compatibility issue when migrating track 2
```

> * Fix the Python SDK compatibility issue when migrating Track 2

Reviewer: ```suggestion
* Fix the Python SDK compatibility issue when migrating Track 2 (#18786)
```

* `az webapp hybrid-connection add`: Improve help/error message and unblock Linux (#18843)
* `az webapp config access-restriction remove`: Fix #18947 issue removing service endpoint rules (#18986)
* : Fix #17424: `az appservice plan show`: Provide correct exit status (#18994)

**ARM**

* `az what-if`: Fix output formatting (#18721)
* `az bicep uninstall`: Add new command to uninstall bicep (#18744)
* `az bicep build`: Fix an issue where running with --stdout doesn't print any output (#18744)
* `az provider register`: Add deprecate info for `--accept-term` (#18739)
* `az lock create/delete`: Add examples for operating different levels of locks (#18890)
* `az deployment group/sub/mg/tenant create`: Add --what-if parameter for invoking What-If with the deployment create commands. (#18924)
* `az deployment group/sub/mg/tenant create`: Add --proceed-if-no-change parameter to skip confirmation when --confirm-with-what-if is set and there's no changes in What-If results. (#18924)
* Bump api-version from 2020-10-01 to 2021-04-01 (#18923)
* `az ts create`: Make parameter `--template-file` support bicep file (#18888)

> * Add example for creating site extension to web app (#18935)

Reviewer: ```suggestion
* `az resource create`: Add example for creating site extension to web app (#18935)
```

* `az ts export`: Fix the issue that export template specs with no linked templates failed (#18928)

**PolicyInsights**

* Migrate to track 2 SDK (#18740)

**Backup**

* `az backup vault`: Add support for Customer Managed Keys(CMK) (#18733)
* `az backup restore restore-disks`: Add MSI usage in IaaS VM Restore (#18961)

**CDN**

* `az cdn endpoint rule`: Add OriginGroupOverride action support (#18711)

**Compute **

> * Fix #18633: az sig image-version create does not support mixing disks, snapshots, and vhd (#18741)

Reviewer: ```suggestion
* `az sig image-version create`: Support mixing disks, snapshots, and vhd (#18741)
```

> * Upgrade package version to fix securityProfile issue in `az vmss update` (#18788)

Reviewer: ```suggestion
* `az vmss update`: Upgrade package version to fix securityProfile issue (#18788)
```

* `az vm boot-diagnostics get-boot-log`: Fix crash when getting boot diagnostics log (#18830)
* `az vm list-skus`: Fix the issue that it can't query the SKU which with partially zones available (#18939)
* `az vm auto-shutdown`: Fix the issue that `--webhook` is required when `--email` is passed in (#18958)
* `az vm create`: Support creating VM from shared gallery image (#19037)
* `az vm secret add`: Add note to use Azure Key Vault VM extension instead in help (#19045)

**Container**

* `az container exec`: Fix and improve terminal experience (#18909)

**DataBoxEdge**

* Migrate databoxedge to track2 SDK (#18678)

**DMS**

> * `az dms project create/az dms project task create`: Removing MySQL projects/tasks for online migrations since they are no longer supported. (#18709)

Reviewer 2: ```suggestion
* `az dms project create/az dms project task create`: Remove MySQL projects/tasks for online migrations since they are no longer supported. (#18709)
```

**IoT**

> * `az iot hub create/update`: Added checks to prevent bad file-upload identity parameters when hub doesn't have identity (#18966)
> * `az iot hub create/update`: Added `--fileupload-notification-lock-duration` parameter (#18966)
> * `az iot hub create/update`: Deprecated `fileupload-storage-container-uri` parameter (#18966)

Reviewer 2: ```suggestion
* `az iot hub create/update`: Add checks to prevent bad file-upload identity parameters when hub doesn't have identity (#18966)
* `az iot hub create/update`: Add `--fileupload-notification-lock-duration` parameter (#18966)
* `az iot hub create/update`: Deprecate `fileupload-storage-container-uri` parameter (#18966)
```

* `az iot dps/hub certificate create`: Certificates will now always be uploaded in base64 encoding. (#18966)

**Key Vault**

* [BREAKING CHANGE] Fix #13752: az keyvault create not idempotent. Creating existing keyvault will fail. (#18520)
* Fix #6372: table output for secrets isn't correct (#18308)

**Maps**

* `az maps creator create`: Support maps creator create managed (#18450)
* `az maps creator update`: Support maps creator update managed (#18450)
* `az maps creator list`: Support maps creator list managed (#18450)
* `az maps creator show`: Support maps creator show managed (#18450)
* `az maps creator delete`: Support maps creator delete managed (#18450)

**NetAppFiles**

> * Update help description for pool-change (#18835)

Reviewer: ```suggestion
* `az netappfiles volume pool-change`: Update help description for pool-change (#18835)
```

**Network**

> * `az network application-gateway create`: Added `--ssl-certificate-name` argument (#18861)

Reviewer 2: ```suggestion
* `az network application-gateway create`: Add `--ssl-certificate-name` argument (#18861)
```

* `az network lb frontend-ip update`: Support cross tenant for argument `--gateway-lb`. (#18792)
* `az network nic ip-config update`: Support cross tenant for argument `--gateway-lb`. (#18792)
* Private link add Microsoft.StorageSync/storageSyncServices provider (#19000)
* Private link add Microsoft.Media/mediaservices provider (#18997)
* Private link add Microsoft.Batch/batchAccounts provider (#18970)

**Packaging**

* Add licenses to all Python packages (#18749)
* Add SOCKS Proxy Support (#18931)

**PolicyInsights**

* Migrate to track 2 SDK (#18740)

**RDBMS**

* PostgreSQL, MySQL migration to GA API (#18921)

**Redis**

> * Add `--redis-version` param to `az redis create` and `az redis update` (#18996)

Reviewer: ```suggestion
* `az redis create\update`: Add new parameter `--redis-version` (#18996)
```

**SQL**

* Update Microsoft.Sql to track2 SDK (#18637)
* `az sql server outbound-firewall-rule create`: Azure CLI Commands for Outbound Firewall Rules (#18671)

**Storage**

* Fix #18352: `az storage fs file list --exclude-dir` breaks with `--show-next-marker` (#18816)
* `az storage fs generate-sas`:  Support generate sas token for file system in ADLS Gen2 account (#18768)
* `az storage account blob-service-properties`: Support last access tracking policy (#18731)

> * Support Version level Worm (VLW) (#18540)

Reviewer: ```suggestion
* `storage container-rm migrate-vlw`: Support Version level Worm (VLW) (#18540)
```

* `az storage copy` add new option `--cap-mbps` (#18344)

**Synapse**

> * Workspace key update failure (#18719)

Reviewer: ```suggestion
* `synapse workspace key update`: Fix the issue that updating a workspace key failure due to parameter `--is-active-cmk` lost (#18719)
```

> * Reimport notebook Failure (#18718)

Reviewer: ```suggestion
* Reimport notebook failure (#18718)
```

Release History
===============

2.27.0
++++++

**ACR**

* [BREAKING CHANGE] `az acr connected-registry install info`: Add a new required parameter `--parent-protocol`. (#18954)
* [BREAKING CHANGE] `az acr connected-registry install renew-credentials`: Add a new required parameter `--parent-protocol`. (#18954)
* `az acr import`: Support new parameter `--no-wait` (#18572)
* Fix the Python SDK compatibility issue when migrating Track 2
* `az acr build`: Make file .dockerignore include directories with `!` (#18821)

**AKS**

* `az aks check-acr`: Fix issues parsing certain client minor versions (#18727)

**AppConfig**

* [BREAKING CHANGE] `appconfig feature set`: Set the value of parameter `--description` to empty string if it is not specified (#18907)
* [BREAKING CHANGE] `az appconfig feature`: Support namespacing for feature flags and change output fields (#18990)

> * Add tags support when creating resource (#18783)

Reviewer: ```suggestion
* Add tags support when creating resource (#18783)
```

Release History
===============

2.27.0
++++++

**ACR**

> * [BREAKING CHANGE] az acr connected-registry install info: Add a new required parameter --parent-protocol. (#18954)
> * [BREAKING CHANGE] az acr connected-registry install renew-credentials: Add a new required parameter --parent-protocol. (#18954)

Reviewer: ```suggestion
* [BREAKING CHANGE] `az acr connected-registry install info`: Add a new required parameter --parent-protocol. (#18954)
* [BREAKING CHANGE] `az acr connected-registry install renew-credentials`: Add a new required parameter --parent-protocol. (#18954)
```

> * `az acr import`: Support no-wait (#18572)

Reviewer: ```suggestion
* `az acr import`: Support new parameter `--no-wait` (#18572)
```

* Fix some bugs on uncovered cmds (#18786)

> * `az acr build`: Make .dockerignore include directories with `!` (#18821)

Reviewer: ```suggestion
* `az acr build`: Make `.dockerignore` include directories with `!` (#18821)
```

> * `az acr build`: Make .dockerignore include directories with `!` (#18821)

Reviewer: ```suggestion
* `az acr build`: Make file .dockerignore include directories with `!` (#18821)
```

**AKS**

* `az aks check-acr`: Fix issues parsing certain client minor versions (#18727)

> **App Config**
> 
> * [BREAKING CHANGE] `appconfig feature set`: Set the value of parameter `--description` to empty string if it is not specified (#18907)
> * [BREAKING CHANGE] `az appconfig feature`: Support namespacing for feature flags and change output fields (#18990)

Reviewer 2: ```suggestion
**AppConfig**

* [BREAKING CHANGE] `appconfig feature set`: Set the value of parameter `--description` to empty string if it is not specified (#18907)
* [BREAKING CHANGE] `az appconfig feature`: Support namespacing for feature flags and change output fields (#18990)
* Add tags support when creating resource (#18783)
```

> **App Config**
> 
> * [BREAKING CHANGE] `appconfig feature set`: Set the value of parameter `--description` to empty string if it is not specified (#18907)
> * [BREAKING CHANGE] `az appconfig feature`: Support namespacing for feature flags and change output fields (#18990)
> 
> **App Service**
> 
> * `az webapp config set`: Add support for VNet Route All property. (#18460)
> * `az webapp vnet-integration add`: Default to VNet Route All. Allow cross subscription integration. (#18460)
> * `az appservice ase create`: Support for ASEv3 External and Zone redundancy (#18748)
> * `az webapp hybrid-connection add`: Improve help/error message and unblock Linux (#18843)
> * `az webapp config access-restriction remove`: Fix #18947 issue removing service endpoint rules (#18986)
> * : Fix #17424: `az appservice plan show`: Provide correct exit status (#18994)
> 
> **AppConfiguration**
> 
> * Add tags support when creating resource (#18783)

Reviewer: ```suggestion
**App Config**

* [BREAKING CHANGE] `appconfig feature set`: Set the value of parameter `--description` to empty string if it is not specified (#18907)
* [BREAKING CHANGE] `az appconfig feature`: Support namespacing for feature flags and change output fields (#18990)
* `az appconfig create`: Add tags support when creating resource (#18783)

**App Service**

* `az webapp config set`: Add support for VNet Route All property. (#18460)
* `az webapp vnet-integration add`: Default to VNet Route All. Allow cross subscription integration. (#18460)
* `az appservice ase create`: Support for ASEv3 External and Zone redundancy (#18748)
* `az webapp hybrid-connection add`: Improve help/error message and unblock Linux (#18843)
* `az webapp config access-restriction remove`: Fix #18947 issue removing service endpoint rules (#18986)
* : Fix #17424: `az appservice plan show`: Provide correct exit status (#18994)
```

> **AppConfiguration**
> 
> * Add tags support when creating resource (#18783)
> 

Reviewer 2: ```suggestion
```

**ARM**

* `az what-if`: Fix output formatting (#18721)
* `az bicep uninstall`: Add new command to uninstall bicep (#18744)
* `az bicep build`: Fix an issue where running with --stdout doesn't print any output (#18744)
* `az provider register`: Add deprecate info for `--accept-term` (#18739)
* `az lock create/delete`: Add examples for operating different levels of locks (#18890)
* `az deployment group/sub/mg/tenant create`: Add --what-if parameter for invoking What-If with the deployment create commands. (#18924)
* `az deployment group/sub/mg/tenant create`: Add --proceed-if-no-change parameter to skip confirmation when --confirm-with-what-if is set and there's no changes in What-If results. (#18924)
* Bump api-version from 2020-10-01 to 2021-04-01 (#18923)
* `az ts create`: Make parameter `--template-file` support bicep file (#18888)
* Add example for creating site extension to web app (#18935)
* `az ts export`: Fix the issue that export template specs with no linked templates failed (#18928)

> **AzurePolicy**

Reviewer 2: ```suggestion
**PolicyInsights**
```

* Migrate to track 2 SDK (#18740)

**Backup**

> * `backup vault`: Add support for Customer Managed Keys(CMK) (#18733)

Reviewer 3: ```suggestion
* `az backup vault`: Add support for Customer Managed Keys(CMK) (#18733)
```

* `az backup restore restore-disks`: Add MSI usage in IaaS VM Restore (#18961)

**CDN**

* `az cdn endpoint rule`: Add OriginGroupOverride action support (#18711)

**Compute **

> * Fix issue #18633: az sig image-version create does not support mixing disks, snapshots, and vhd (#18741)

Reviewer 3: ```suggestion
* Fix #18633: az sig image-version create does not support mixing disks, snapshots, and vhd (#18741)
```

* Upgrade package version to fix securityProfile issue in `az vmss update` (#18788)
* `az vm boot-diagnostics get-boot-log`: Fix crash when getting boot diagnostics log (#18830)
* `az vm list-skus`: Fix the issue that it can't query the SKU which with partially zones available (#18939)
* `az vm auto-shutdown`: Fix the issue that `--webhook` is required when `--email` is passed in (#18958)
* `az vm create`: Support creating VM from shared gallery image (#19037)

> * `vm secret add`: Add note to use Azure Key Vault VM extension instead in help (#19045)

Reviewer 3: ```suggestion
* `az vm secret add`: Add note to use Azure Key Vault VM extension instead in help (#19045)
```

**NetAppFiles**

* Update help description for pool-change (#18835)

**Network**

* `az network application-gateway create`: Added `--ssl-certificate-name` argument (#18861)
* Private link add Microsoft.ServiceBus/namespaces provider (#18999)
* `az network application-gateway waf-policy custom-rule match-condition add`: Add examples (#18957)
* `az network express-route port link update`: Add `--macsec-sci-state` argument. (#18814)
* Private link add Microsoft.Web/hostingEnvironments provider (#19025)
* `az network lb frontend-ip update`: Support cross tenant for argument `--gateway-lb`. (#18792)
* `az network nic ip-config update`: Support cross tenant for argument `--gateway-lb`. (#18792)
* Private link add Microsoft.StorageSync/storageSyncServices provider (#19000)
* Private link add Microsoft.Media/mediaservices provider (#18997)
* Private link add Microsoft.Batch/batchAccounts provider (#18970)

**Packaging**

* Add licenses to all Python packages (#18749)
* Add SOCKS Proxy Support (#18931)

**RDBMS**

* PostgreSQL, MySQL migration to GA API (#18921)

**Redis**

* Add `--redis-version` param to `az redis create` and `az redis update` (#18996)

**SQL**

> * BREAKING CHANGES: (#18637)

Reviewer 3: ```suggestion
* Update Microsoft.Sql to track2 SDK (#18637)
```
